pass-data-client/src/test/java/org/eclipse/pass/support/client/SubmissionStatusCalculatorTest.java

```diff
@@ -0,0 +1,571 @@
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED_NEWUSER));
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED_NEWUSER),
+                          submissionEvent(ZonedDateTime.now(), EventType.CHANGES_REQUESTED),
+                          submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED));
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        assertEquals(SubmissionStatus.APPROVAL_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, SubmissionStatus.DRAFT));
+    }
+
+    /**
+     * Tests various scenarios in which the pre-submission status should be "CHANGES_REQUESTED"
+     */
+    @Test
+    public void testPreSubmissionStatusChangesRequested() {
+        List<SubmissionEvent> submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.CHANGES_REQUESTED));
+        assertEquals(SubmissionStatus.CHANGES_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED),
+                          submissionEvent(ZonedDateTime.now(), EventType.CHANGES_REQUESTED));
+        assertEquals(SubmissionStatus.CHANGES_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        submissionEvents =
+            Arrays.asList(submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED),
+                          submissionEvent(ZonedDateTime.now(), EventType.CHANGES_REQUESTED),
+                          submissionEvent(ZonedDateTime.now(), EventType.APPROVAL_REQUESTED),
+                          submissionEvent(ZonedDateTime.now(), EventType.CHANGES_REQUESTED));
+        assertEquals(SubmissionStatus.CHANGES_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, null));
+
+        assertEquals(SubmissionStatus.CHANGES_REQUESTED,
+                     SubmissionStatusCalculator.calculatePreSubmissionStatus(submissionEvents, SubmissionStatus.DRAFT));
+    }
+
+    /**
+     * Tests various scenarios in which the pre-submission status should be "CANCELLED"
+     */
+    @Test
+    public void testPreSubmissionStatusCancelled() {
```

Comment:
When doing a mvn clean install, I'm getting this test failure:

[ERROR] testPreSubmissionStatusCancelled  Time elapsed: 0.001 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <CANCELLED> but was: <APPROVAL_REQUESTED>
        at org.eclipse.pass.support.client.SubmissionStatusCalculatorTest.testPreSubmissionStatusCancelled(SubmissionStatusCalculatorTest.java:469)